src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/LinkFieldDisplayDriver.cs

```diff
@@ -81,8 +84,16 @@ public override async Task<IDisplayResult> UpdateAsync(LinkField field, IUpdateM
                     updater.ModelState.AddModelError(Prefix, S["The text default value is required for {0}.", context.PartFieldDefinition.DisplayName()]);
                 }
 
-                // Run this through a sanitizer in case someone puts html in it.
-                // No settings.
+                if (!string.IsNullOrEmpty(field.Url) || !string.IsNullOrEmpty(field.Text))
+                {
+                    // Test the input against the sanitizer.
+                    var aTag = @"<a href=""" + (field.Url ?? "") + @""">" + (field.Text ?? "") + "</a>";
```

Comment:
I am afraid that any special will break this.

For instance Url and Text are not html encoded here, while the view I think they will, and a sanitizer would happen after html encoding. Like have a text with `<script>` would be fine in theory, because it's a text that will be encoded, but I think that this code right now will prevent it.

Reply:
I suspect you are correct. 
Quickly done, so I need to go back and look for a better example of an actual attack point here (if there is one, I'm not totally sure)

I'll mark this not ready and finish up some research first

Reply:
Maybe this field can't be protected. Or the validation should only be based on white list. You can also check what WP does.